### PR TITLE
Add a new job for collecting table stats

### PR DESCRIPTION
**Pull request by maluchari**
## Summary

This PR introduces a new spark app TableStatsCollectionSparkApp  that collects the stats required for building a system table with iceberg stats and other storage stats as required.

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [X] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [X] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.
- OperationTask identified
```
`2024-04-11 02:50:36 INFO  JobsScheduler:78 - Found operation task class: com.linkedin.openhouse.jobs.scheduler.tasks.OrphanTableDirectoryDeletionTask
`
```
Job tries to fetch a task list
`2024-04-11 02:50:36 INFO  JobsScheduler:131 - Fetching task list based on the job type: TABLE_STATS_COLLECTION`

Job launched for the table that was previously created
`2024-04-11 02:50:38 INFO  OperationTask:97 - Launched a job with id TABLE_STATS_COLLECTION_db_tb_ab0b838c-bfaf-49bf-875b-cce83a7f6181 for dbName: db, tableName: tb, creator: openhouse`

From spark logs:

`INFO util.TableStatsCollector: {\"totalReferencedDataFilesSizeInBytes\":459,\"numReferencedDataFiles\":1,\"totalDirectorySizeInBytes\":16955,\"numObjectsInDirectory\":5,\"currentSnapshotId\":834391413618423431,\"currentSnapshotTimestamp\":1712812466418,\"numCurrentSnapshotReferencedDataFiles\":1,\"totalCurrentSnapshotReferencedDataFilesSizeInBytes\":459,\"oldestSnapshotTimestamp\":1712812466418,\"numExistingMetadataJsonFiles\":2,\"numReferencedManifestFiles\":1,\"numReferencedManifestLists\":1,\"recordTimestamp\":1712813575606,\"clusterName\":\"LocalHadoopCluster\",\"databaseName\":\"db\",\"tableName\":\"tb\",\"tableUUID\":\"19f47c31-dc3d-4549-85c9-6c5e27a6c614\",\"tableLocation\":\"/data/openhouse/db/tb-19f47c31-dc3d-4549-85c9-6c5e27a6c614\",\"tableCreator\":\"openhouse\",\"tableCreationTimestamp\":0,\"tableLastUpdatedTimestamp\":0,\"tableType\":\"PRIMARY_TABLE\"}"`

snapshotId matches with below query

```
scala> spark.sql("select * from openhouse.db.tb.snapshots").show()
+--------------------+------------------+---------+---------+--------------------+--------------------+
|        committed_at|       snapshot_id|parent_id|operation|       manifest_list|             summary|
+--------------------+------------------+---------+---------+--------------------+--------------------+
|2024-04-11 05:14:...|834391413618423431|     null|   append|/data/openhouse/d...|{spark.app.id -> ...|
+--------------------+------------------+---------+---------+--------------------+--------------------+
```
An example where data file size matches to verify storage stats.
("totalCurrentSnapshotReferencedDataFilesSizeInBytes\":459 from above)

```
openhouse@4a7eff5072d5:/opt/spark$ hdfs dfs -ls /data/openhouse/db/tb-19f47c31-dc3d-4549-85c9-6c5e27a6c614/data/ts_day=2024-03-12
Found 1 items
-rw-r--r--   3 openhouse supergroup        459 2024-04-11 05:14 /data/openhouse/db/tb-19f47c31-dc3d-4549-85c9-6c5e27a6c614/data/ts_day=2024-03-12/00000-0-341233a9-2dea-490c-a059-584d3a1cb84f-00001.orc
```

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/TableStatsCollectionSparkApp.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  @Override
+  protected void runInner(Operations ops) {
+    log.info("Running TableStatsCollectorApp for table {}", fqtn);
+
+    ops.collectTableStats(fqtn);
+  }
+
+  public static void main(String[] args) {
+    List<Option> extraOptions = new ArrayList<>();
+    extraOptions.add(new Option("t", "tableName", true, "Fully-qualified table name"));
```
Comment (sumedhsakdeo) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/TableStatsCollectionSparkApp.java:32]: Are we using tableName, or fqtn as a standard in other job types? Let's make sure we are consistent.

**Review by sumedhsakdeo — CHANGES_REQUESTED** [review]
Clean PR @maluchari , can you also update testing done section with prepared event logged, ie show it works on a table in local docker.

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/TableStatsCollectionSparkApp.java (hunk 1 above)
Comment (maluchari) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/TableStatsCollectionSparkApp.java:32]: This usage is consistent with other jobs that we have today. https://github.com/linkedin/openhouse/blob/main/apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/SnapshotsExpirationSparkApp.java#L83

**Comment by maluchari** [comment]
> Clean PR @maluchari , can you also update testing done section with prepared event logged, ie show it works on a table in local docker.

Updated with spark application logs

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public JobsScheduler(
   }
 
   public static void main(String[] args) {
-    log.info("Starting scheduler");
+    log.info("Starting scheduler Updated");
```
Comment (rohitkum2506) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java]: What's the relevance of "Updated"?

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/TableStatsCollector.java [hunk 3]
```diff
@@ -44,6 +45,6 @@ public IcebergTableStats collectAndPublishTableStats() {
    */
   void publishStats(IcebergTableStats stats) {
     log.info("Publishing stats for table: {}", fqtn);
-    log.info("Stats: {}", stats);
+    log.info(new Gson().toJson(stats));
```
Comment (maluchari) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/TableStatsCollector.java:48]: Updated this to print the stats properly

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java (hunk 2 above)
Comment (maluchari) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java]: This was an unintended change that came in as part of an old commit. Removed in the subsequent ones.

**Review by sumedhsakdeo — APPROVED** [review]
💯

**Review by teamurko — APPROVED** [review]

**Review by rohitkum2506 — APPROVED** [review]

**Pull request merged**